src/routes.rs

```diff
@@ -1922,3 +2057,338 @@ pub(crate) async fn unlock(
+        let unlocked_state = state.check_unlocked().await?.clone().unwrap();
+        let ldk_data_dir_path = PathBuf::from(state.static_state.ldk_data_dir.clone());
+
+        let swapstring = SwapString::from_str(&payload.swapstring).map_err(|_| APIError::InvalidSwapString(payload.swapstring.clone()))?;
+        let payment_secret = hex_str_to_vec(&payload.payment_secret)
+            .and_then(|data| data.try_into().ok())
+            .map(|data| PaymentSecret(data))
+            .ok_or(APIError::InvalidArgument(s!("Invalid hex string for payment_secret")))?;
+        let taker_pk = bitcoin::secp256k1::PublicKey::from_str(&payload.taker_pubkey).map_err(|_| APIError::InvalidPubkey)?;
+
+        if get_current_timestamp() > swapstring.expiry {
+            return Err(APIError::Expired);
+        }
+
+        let payment_preimage = unlocked_state.channel_manager.get_payment_preimage(swapstring.payment_hash, payment_secret)
+            .map_err(|_| APIError::InvalidArgument(s!("Unable to find payment preimage, have you provided the correct swapstring and payment secret?")))?;
+
+        // The cli takes the swaptype from the point of view of the user (=who wants to do the trade)
+        // In this function we're the market maker (=who sends the payment, completing the user trade)
+        // We swap the swaptype to hopefully make this function easier to read
+        let swaptype = swapstring.swap_type.opposite();
+
+        let receive_hints = unlocked_state.channel_manager
+            .list_usable_channels()
+            .iter()
+            .map(|details| {
+                let config = details.counterparty.forwarding_info.as_ref().unwrap();
+                RouteHint(vec![RouteHintHop {
+                    src_node_id: details.counterparty.node_id,
+                    short_channel_id: details.short_channel_id.unwrap(),
+                    cltv_expiry_delta: config.cltv_expiry_delta,
+                    htlc_maximum_msat: None,
+                    htlc_minimum_msat: None,
+                    fees: RoutingFees {
+                        base_msat: config.fee_base_msat,
+                        proportional_millionths: config.fee_proportional_millionths,
+                    },
+                }])
+            })
+            .collect();
+
+        let first_leg = get_route(
+            &unlocked_state.channel_manager,
+            &unlocked_state.router,
+            unlocked_state.channel_manager.get_our_node_id(),
+            taker_pk,
+            if swaptype.is_buy() { Some(swaptype.amount_msats()) } else { None },
+            if swaptype.is_buy() { None } else { Some(swapstring.asset_id) },
+            vec![],
+        );
+        let second_leg = get_route(
+            &unlocked_state.channel_manager,
+            &unlocked_state.router,
+            taker_pk,
+            unlocked_state.channel_manager.get_our_node_id(),
+            if swaptype.is_buy() { Some(HTLC_MIN_MSAT) } else { Some(swaptype.amount_msats()) },
```

Comment:
Because the default may change in the future and it's better not to rely on these assumptions in my opinion. The default is for when I don't know what value to use, so I let the function decide for me.

Here I know exactly what value I want, so I'm passing it in even if I know the default (right now) is the same.

Reply:
I don't understand. Why in the `first_leg` route you're ok in having a value that could change if the default value in `get_route` changes? IMO we can change the `get_route` signature to always get a `final_value_msat` (so making it mandatory instead of optional)